### PR TITLE
iOS Test target

### PR DESCRIPTION
**Pull request by phatblat**
iOS Tests #444 

screen shot 2015-07-10 at 1 25 16 am

To help prevent things like #478 & #481.

Using [SSZipArchive](https://github.com/ZipArchive/ZipArchive) isn't ideal since it isn't a framework so the Xcode integration is ugly. But at least we can now run tests on iOS. Boy, are they slow. :zzz:

### CODE REVIEW

**Comment by pietbrauer** [comment]
:+1: Travis [recently introduced](http://blog.travis-ci.com/2015-07-08-xcode-6-4-now-available/) longer build times on Mac machines (120 minutes). Maybe we can get public CI working again then? What do you think @joshaber?

**Comment by joshaber** [comment]
Sure, we can try Travis again if someone wants to set it up :+1:

**Comment by pietbrauer** [comment]
✋ I volunteer, progress tracked in #485

**Comment by pietbrauer** [comment]
@phatblat if you resolve the merge conflicts and push again Travis will be started :wink:

**Comment by phatblat** [comment]
Another PR build is green! Ready for review.

File: ObjectiveGitTests/QuickSpec+GTFixtures.m [hunk 1]
```diff
@@ -95,6 +99,24 @@ - (NSString *)pathForFixtureRepositoryNamed:(NSString *)repositoryName {
...
 - (BOOL)unzipFile:(NSString *)member fromArchiveAtPath:(NSString *)zipPath intoDirectory:(NSString *)destinationPath error:(NSError **)error {
+
+#if TARGET_OS_IPHONE
+	// iOS: unzip in-process using SSZipArchive
+	//
+	// system() and NSTask() are not available when running tests in the iOS simulator
+
+	BOOL success = [SSZipArchive unzipFileAtPath:zipPath toDestination:destinationPath overwrite:YES password:nil error:error];
+
+    if (!success) {
```
Comment (joshaber) [inline ObjectiveGitTests/QuickSpec+GTFixtures.m]: Nit: it looks like there's a mix of tabs/spaces here.

File: ObjectiveGitTests/QuickSpec+GTFixtures.m [hunk 2]
```diff
@@ -95,6 +99,24 @@ - (NSString *)pathForFixtureRepositoryNamed:(NSString *)repositoryName {
 }
 
 - (BOOL)unzipFile:(NSString *)member fromArchiveAtPath:(NSString *)zipPath intoDirectory:(NSString *)destinationPath error:(NSError **)error {
+
+#if TARGET_OS_IPHONE
+	// iOS: unzip in-process using SSZipArchive
```
Comment (joshaber) [inline ObjectiveGitTests/QuickSpec+GTFixtures.m]: Any reason not to use SSZipArchive on both?

**Comment by joshaber** [comment]
Really fantastic work! Just a couple nits.

File: ObjectiveGitTests/QuickSpec+GTFixtures.m (hunk 2 above)
Comment (phatblat) [inline ObjectiveGitTests/QuickSpec+GTFixtures.m]: That would certainly make things cleaner. My only concern is speed. The iOS tests are quite a bit slower than the Mac tests and I'm not sure why. I'll experiment with using SSZipArchive for the Mac tests to see if they slow down or not.

**Comment by phatblat** [comment]
Using SSZipArchive is almost an order of magnitude slower than shelling out to unzip:

- MacTests (unzip): 1:20
- iOSTests (SSZipArchive): 10:49
- MacTests (SSZipArchive): 10:35

I'm inclined to leave the Mac tests using unzip. I'd like to see the iOS tests run faster; perhaps once that's solved we could switch the Mac tests to use the same unzip mechanism.

**Comment by joshaber** [comment]
Wow, that's nuts! Thanks for doing the legwork on that.

:metal:

**Pull request merged**